src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,285 @@
+    rels_mondo_source = set()
+    rels_mondo_non_source = set()
+    rels_mondo_non_synced = set()
+    # is_a edges queried from Mondo
+    # - rels_mondo_source_raw: where both parent/child are from source
+    # - rels_mondo_non_source: where parent and/or child not from source
+    # - rels_mondo_non_synced: where parent and/or child Mondo ID exists in mondo.db but not mondo.sssom.tsv
+    for x in rels_mondo_mondo:
+        try:
+            rels_mondo_source.add((mondo_source_map[x[0]], x[1], mondo_source_map[x[2]]))
+        except KeyError:
+            try:
+                rels_mondo_non_source.add((mondo_source_map_all[x[0]], x[1], mondo_source_map_all[x[2]]))
+            # TODO: PR comment
+            # TODO: For now, print this at end, but when this work is being completed, make this throw an err
+            except KeyError:
+                rels_mondo_non_synced.add(x)
+    # - alterantive: will only work if mondo.db and mondo.sssom.tsv are in sync
+    # rels_mondo_source_raw = [(mondo_source_map_all[x[0]], x[1], mondo_source_map_all[x[2]]) for x in rels_mondo_mondo]
+    # rels_mondo_source = [x for x in rels_mondo_source_raw if any([x[0].startswith(y) for y in owned_prefix_map]) and any([x[2].startswith(y) for y in owned_prefix_map])]
+    t1 = datetime.now()
+    print(f'  Completed in {(t1 - t0).seconds} seconds')
+
+    # TODO: set diffs to produce these categories
+    #    1. SCR direct in source and Mondo
+    #      3. SCR is direct in the source, but not at all in Mondo
+    #        10. SCR is excluded in Mondo, but direct in source
+    #      5. SCR is direct in Mondo, and not in the source at all.
+
+    # TODO: change this to proper robot template
+    # sub / obj / is_direct / is_excluded / in_mondo / in_source
+    rows = []
+    in_both = rels_source.intersection(rels_mondo_source)
+    in_source = rels_source.difference(rels_mondo_source)
+    in_mondo = rels_mondo_source.difference(rels_source)
+    # todo: this could be DRYer
+    for rel in in_both:
+        rows.append({
+            'subject': rel[0],
+            'object': rel[2],
+            'is_direct': True,
+            'is_excluded': 'tbd',  # todo
+            'in_mondo': True,
+            'in_source': True,
+        })
+    for rel in in_source:
+        rows.append({
+            'subject': rel[0],
+            'object': rel[2],
+            'is_direct': True,
+            'is_excluded': 'tbd',  # todo
+            'in_mondo': False,
+            'in_source': True,
+        })
+    for rel in in_mondo:
+        rows.append({
+            'subject': rel[0],
+            'object': rel[2],
+            'is_direct': True,
+            'is_excluded': 'tbd',  # todo
+            'in_mondo': True,
+            'in_source': False,
+        })
+    df = pd.DataFrame(rows)
+    df.to_csv(str(REPORTS_DIR / f'{EX_ONTO_NAME}-subclassTempReport.tsv'), sep='\t')
```

Comment:
## Update 2023/10/25
Might want to see if we can clear some of the other threads. PR commentary very cluttered. Just gonna keep everything in 1 comment today.

I had a busy last 2-3 weeks with OHDSI Symposium and broken computer. I had 2-3 hours today to work on this. Better progress than expected.

I created a temporary output format with the following columns:
`sub / obj / is_direct / is_excluded / in_mondo / in_source`
[ordo-subclassTempReport.tsv.zip](https://github.com/monarch-initiative/mondo-ingest/files/13172110/ordo-subclassTempReport.tsv.zip)

The algorithm I've employed is basically like this:
1. Get all terms for given source, as they appear in source, from `mondo-ingest.db`
2. Get all terms from source, as they appear in Mondo, from `mondo.sssom.tsv`
3. Get all SC relationships as they appear in source from `mondo-ingest.db`
4. Get all SC relationships as they appear in mondo from `mondo.db`
5. Determine which are in either / both sets, and write a report.

Also, I found some terms that were in `mondo.db` but not in `mondo.sssom.tsv`.  This has to do with the fact that on this backup computer I can't create many artefacts (including SemanticSQL) due to limited memory, so I got a `mondo.db` from Harshad which is out of sync with my `mondo.sssom.tsv`.

Reply:
This is not yet an "algorithm" but your list of sources look exactly right to me!

I would kinda like two columns added to your output:

subject_mondo_id, object_mondo_id, subject_mondo_label, object_mondo_label

Thanks!